### DISCUSSION TITLE
How to make prerequest scripts generated for Variation Tests?

### DISCUSSION THREAD

**Discussion by armen-ch** [apideck-libraries/portman#477] [Q&A]
Hello,
How can I define "prerequest" scripts in the Portman configuration that will be generated in the Postman file for Variations tests?
I have created a Portman config for Variation tests and would like to run a token request before each request in the Variation tests in order to use the resulting token in the Variation test request. I'm using the `"operationPreRequestScripts"` option, which looks something like this:

   ```
 "operationPreRequestScripts": [
    {
      "openApiOperation": "*::/products*",
      "scripts": [
        "file:scripts/get_token.js"
      ]
    },
    {
      "openApiOperation": "*::/system/products*",
      "scripts": [
        "file:scripts/get_system_token.js"
      ]
    }
  ]
```
The above configuration generates `"listening": "prerequest"` events with injected script content from the specified file(s) for Contract tests. But there are no` "listening": "prerequest"` events generated for Variation tests.
Please advise.

**Comment by armen-ch** [comment]
hi @thim81, Do you have solution for this case? This really blocked me. And I can't move forward. Actually I see couple of issues here. First, I think we globally defined Pre-Request Scripts should be applied to all the tests Contract, Variation etc. until they are overwritten with the ones defined in lower levels. 
Another one is that "operationPreRequestScripts" defined in test level doesn't work properly. It fails to apply OpenAPI reference pattern to the right target correctly. For this case I created issue #478. I'm doing some OAS to Postman POC with Portman and my variation test can't make request with token since I can't run correct pre-request scripts token provider service.
Please advise.

**Reply by thim81** [reply]
Hi @armen-ch,

I provided a workaround as a response to #478 
We need more time to look into the desired behavior for Variations where there is targeting inside a variation.

About the behavior of global configs like Pre-Request Scripts, overwrites, assignVariables should be applied globally to all tests + variations, have been mentioned in a similar request (#360).

I'm going to consult with @nicklloyd, on how to tackle the globals being pushed to variations as well.
It could be a setting on the global configs, like "inherit: true" or "applyGlobal:true"

```json
"overwrites": [
    {
      "openApiOperation": "*::/(v1|v2)/products*",
      "inherit": true,
      "overwriteRequestHeaders": [
        {
          "key": "Authorization",
          "value": "Bearer {{businessToken}}",
          "insert": true
        }
      ]
    },
    {
      "openApiOperation": "*::/(v1|v2)/system/*",
      "inherit": false,
      "overwriteRequestHeaders": [
        {
          "key": "Authorization",
          "value": "Bearer {{systemToken}}",
          "insert": true
        }
      ]
    }
  ]
```

**Comment by armen-ch** [comment]
Thank you @thim81 for taking this into consideration. Having configuration parameter for down forcing global settings would be very useful addition to Portman, I think. Also I posted my expectation with regards of targeting priorities in Portman configuration blocks in [#478](https://github.com/apideck-libraries/portman/issues/478)
